pulsar-client-cpp/lib/PartitionedConsumerImpl.cc

```diff
@@ -389,20 +371,13 @@ void PartitionedConsumerImpl::notifyResult(CloseCallback closeCallback) {
     }
 }
 
-void PartitionedConsumerImpl::setState(const PartitionedConsumerState state) {
-    Lock lock(mutex_);
-    state_ = state;
-    lock.unlock();
-}
+void PartitionedConsumerImpl::setState(const PartitionedConsumerState state) { state_ = state; }
```

Comment:
I think we don't need a `setState` method now with the atomic state. It's better to remove the `setState` method and use `state_ = xxx` instead.